### PR TITLE
Fixes: Poetry install module fixes

### PR DESCRIPTION
**Pull request by DerekMaggio**
# Changelog

- Add version specification e.g. `find_package(Poetry 1.1.14)`
- Move poetry install stuff to `stm32-tools`
- Set poetry executable path to be `ot3-firmware/stm32-tools/poetry/bin/poetry`
- Output `Poetry_EXECUTABLE` variable that contains above executable path

### CODE REVIEW

**Comment by DerekMaggio** [comment]
@fsinapi don't hurt yourself with this one

File: FindPoetry.cmake [hunk 1]
```diff
@@ -19,7 +19,7 @@ else()
 endif()
 
 
-set(LOCALINSTALL_POETRY_DIR "${CMAKE_SOURCE_DIR}/poetry")
+set(LOCALINSTALL_POETRY_DIR "${CMAKE_SOURCE_DIR}/stm32-tools/poetry")
 message(STATUS "Downloading poetry install script to: ${LOCALINSTALL_POETRY_DIR}")
 
 message(STATUS "Installing Poetry")
```
Comment (fsinapi) [inline FindPoetry.cmake:25]: Missed this in the last PR - the `PREFIX` in `FetchContent_Declare` should actually just be `${CMAKE_SOURCE_DIR}/stm32-tools/`, and then the SOURCE_DIR stays the same.

**Review by fsinapi — APPROVED** [review]
Good to go with one tiny change I missed before (sorry!)

File: FindPoetry.cmake (hunk 1 above)
Comment (DerekMaggio) [reply FindPoetry.cmake:25]: I think your suggestion and my current implementation are incorrect.

What I get with your suggestion is:
```
stm32-tools 
  poetry
    src
    tmp
    install_poetry.py
  src
  tmp
```

What I get with my current implementation is:
```
stm32-tools 
  poetry
    src
      install_poetry.py
    tmp
    install_poetry.py
```

What I want is:
```
stm32-tools
  poetry
    install_poetry.py 
```    

What I actually need to do is remove `PREFIX`

**Review by sfoster1 — APPROVED** [review]
Yeah, that looks great to me if it works. Nice!

**Pull request merged**